### PR TITLE
Trying this; will Redis Cloud initialize as per the docs?

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java [hunk 1]
```diff
@@ -118,6 +112,33 @@ public void run() {
...
+        }
+        if (url != null) {
+            url = config.getProperty("redis.url");
+        }
+        return url;
+    }
+    
+    private JedisPool constructJedisPool(final URI redisURI, final BridgeConfig config) {
+        if (config.isLocal()) {
+            return new JedisPool(new JedisPoolConfig(),
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java]: We're missing the poolConfig.setMaxTotal() call here and below.

File: app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java [hunk 2]
```diff
@@ -82,29 +83,22 @@ public BridgeConfig bridgeConfig() {
     }
 
     @Bean(name = "jedisPool")
-    public JedisPool jedisPool(final BridgeConfig config) {
-
+    public JedisPool jedisPool(final BridgeConfig config) throws Exception {
         // Configure pool
         final JedisPoolConfig poolConfig = new JedisPoolConfig();
         poolConfig.setMaxTotal(config.getPropertyAsInt("redis.max.total"));
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java:89]: This pool config isn't being used. Should it?

File: app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java [hunk 3]
```diff
@@ -82,29 +83,22 @@ public BridgeConfig bridgeConfig() {
...
         // Test pool
         try (Jedis jedis = jedisPool.getResource()) {
             final String result = jedis.ping();
             if (result == null || !"PONG".equalsIgnoreCase(result.trim())) {
-                throw new MissingResourceException(
-                        "No PONG from PINGing Redis: " + result + ".",
-                        JedisPool.class.getName(),
-                        host + ":" + port);
+                throw new MissingResourceException("No PONG from PINGing Redis: " + result + ".",
+                        JedisPool.class.getName(), redisURI.toString());
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java]: Could this leak our Redis credentials into the logs?

File: app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java (hunk 2 above)
Comment (alxdarksage) [inline app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java:89]: Definitely. Now passing it to the JedisPool constructor, and also changed logging NOT to log the entire URL we're getting from Heroku.

**Pull request merged**